### ISSUE TITLE
[Tutorials] loss.backward() called on integer?

### ISSUE THREAD

**Issue by smarianimore** [Farama-Foundation/Gymnasium#741]
### Question

I'm fairly new to Deep RL but I suspect something is wrong in the "Building an agent" section of the tutorial "Training using REINFORCE for Mujoco" ([here](https://gymnasium.farama.org/tutorials/training_agents/reinforce_invpend_gym_v26/#building-an-agent)).
Function "update()" in the provided snippet contains the following lines (removed non relevant ones):
```
loss = 0
[...]
loss.backward()
```
that triggers an obvious warning on most IDEs: there is no backward() attribute (ie method) for integers...
However, I tested the codebase and it appears to be working, also producing a graph similar to the one shown...

Hence my question: what am I missing about that "weird" (apparently, at least to me) call to loss.backward() when loss is an integer?

**Comment by pseudo-rnd-thoughts** [comment]
Thanks for pointing this out, we can change to `torch.Tensor(0)` (or similar that doesn't produce a warning) for an IDE
@smarianimore Would you be interested in doing this?

**Comment by smarianimore** [comment]
@pseudo-rnd-thoughts Im afraid I don’t have the necessary skill yet…I’m an absolute beginner with PyTorch. 

Indeed my confusion is not so much about the warning *per se* but more about 

    1. calling .backwards(), that to my understanding tells PyTorch to compute the gradient of tensors, on an integer (that is not a tensor, so I assume the call silently fails?)
    2. How can the example still seem to work properly even if the call above makes no sense?

I’m sure I’m missing something, but I cannot put my finger on what :/

**Comment by robertoschiavone** [comment]
That's exactly it. `log_prob.mean() * delta * (-1)` is a Tensor, and the Tensor class probably overloads [`__radd__`](https://docs.python.org/3/reference/datamodel.html#object.__radd__). This means that if the left operand, in this case an integer, doesn't support an operation with the right one, Python still checks if the right one supports it, w.r.t. the left one.

To suppress the warning you can simply annotate the type when you declare the `loss` variable, like this

```python
loss: int | Tensor = 0
```

Your IDE now know that the variable can be either type.

**Comment by smarianimore** [comment]
Many thanks, thus no issue with the codebase, only with my ignorance.
This can be closed.
